docs/how-to-guides/themes/theme-json.md

```diff
@@ -464,6 +464,19 @@ There's a growing need to add more theme metadata to the theme.json. This sectio
 }
 ```
 
+**templateParts**: within this field themes can list the template parts present in the `block-template-parts` folder. For example, for a template part named `my-template-part.html`, the `theme.json` can declare what area that template part is intended for to use the corresponding block variation in the editor.  Currently block variations exist for "header" and "footer" values of the area term, any other values and undefined template parts will default to the general template part block.
```

Comment:
I think it needs a bit more clarification about why this has to be set here (`theme.json`) and not as an attribute in the markup. Maybe something like this:
` the Template part block renders its own entity internally. Any attributes on the outer block itself are just block attributes, and thus saved as part of the post it is a child of (template, page, etc).` - you have written this :).

Also a bit more info about what a match with a block variation means (results), like the proper display information and where else will be used for (semantic).

Reply:
> do we need to declare all created Template parts in theme.json or we have to, only if we want to utilize the Block variations API?

We don't have to declare them, as they would default to the general/uncategorized area.  Thats part of what I was getting at with:

> undefined template parts will default to the general template part block.

<hr />

>I think it needs a bit more clarification about why this has to be set here (theme.json) and not as an attribute in the markup. 

I was trying to keep it light, describing what needs to be done and how to do it without adding all the weight of internal database structures, entities, taxonomy terms, etc. etc.  Im happy to add more if people think it is necessary, but I was trying to avoid making the description a bit too heavy and confusing.

Reply:
> and not as an attribute in the markup.

Thinking about it more, especially since we recently added that 'area' attribute used in placeholder/creation flows it probably makes sense to clarify that here. 👍

Reply:
updated a bit 😁